skopt/utils.py

```diff
@@ -501,3 +499,104 @@ def normalize_dimensions(dimensions):
+    dim2 = Real(name='bar', low=0.0, high=1.0)
+    dim3 = Real(name='baz', low=0.0, high=1.0)
+
+    # Gather the search-space dimensions in a list.
+    dimensions = [dim1, dim2, dim3]
+
+    # Define the objective function with named arguments
+    # and use this function-decorator to specify the search-space dimensions.
+    @wrap_named_args(dimensions=dimensions)
+    def my_objective_function(foo, bar, baz):
+        return foo ** 2 + bar ** 4 + baz ** 8
+
+    # Now the function is callable from the outside as
+    # `my_objective_function(x)` where `x` is a list of unnamed arguments,
+    # which then wraps your objective function that is callable as
+    # `my_objective_function(foo, bar, baz)`.
+    # The conversion from a list `x` to named parameters `foo`, `bar`, `baz`
+    # is done automatically.
+
+    # Run the optimizer on the wrapped objective function which is called as
+    # `my_objective_function(x)` as expected by `forest_minimize()`.
+    result = forest_minimize(func=my_objective_function, dimensions=dimensions,
+                             n_calls=20, base_estimator="ET", random_state=4)
+
+    # Print the best-found results.
+    print("Best fitness:", result.fun)
+    print("Best parameters:", result.x)
+
+    Parameters
+    ----------
+    * `dimensions` [list(Dimension)]:
+        List of `Dimension`-objects for the search-space dimensions.
+
+    Returns
+    -------
+    * `wrapped_func` [callable] 
+        Wrapped objective function.
+    """
+
+    def decorator(func):
+        """
+        This uses more advanced Python features to wrap `func` using a
+        function-decorator. To get explanations of how this works in Python,
+        please do an internet search for e.g.:
+        Python function decorator with additional arguments.
+        """
+
+        # Ensure all dimensions are correctly typed.
+        if not all(isinstance(dim, Dimension) for dim in dimensions):
+            raise ValueError("All dimensions must be instances of the Dimension-class.")
+
+        # Ensure all dimensions have names.
+        if any(dim.name is None for dim in dimensions):
+            raise ValueError("All dimensions must have names.")
+
+        @wraps(func)
+        def wrapper(x):
+            # Ensure the number of dimensions match
+            # the number of parameters in the list x.
+            if len(x) != len(dimensions):
+                msg = "Mismatch in number of search-space dimensions. " \
+                      "len(dimensions)=={} and len(x)=={}"
+                msg = msg.format(len(dimensions), len(x))
+                raise ValueError(msg)
+
+            # Create a dict where the keys are the names of the dimensions
+            # and the values are taken from the list of parameters x.
+            arg_dict = {dim.name: value for dim, value in zip(dimensions, x)}
+
+            # Call the wrapped objective function with the named arguments.
+            objective_value = func(**arg_dict)
```

Comment:
I am not convinced the comments above (line 591, 592 and 595) bring anything helpful except repeating the semantics of the code.

Reply:
You are wrong. These comments explain *in English* what is actually fairly unusual Python programming techniques that certainly won't be understood very well by people who are not Python experts. Once again, this code is *not* written for *you*!